src/VASSAL/build/module/Chatter.java

```diff
@@ -21,477 +21,636 @@
-    doc.getText(p0, p1 - p0, s);
-    g.setColor(getColor(elem));
-
-    final Graphics2D g2d = (Graphics2D) g;
-    g2d.addRenderingHints(SwingUtils.FONT_HINTS);
-    return Utilities.drawTabbedText(s, x, y, g, ex, p0);
-  }
-
-  private class WrappedView extends WrappedPlainView {
-    private WrappedView(Element el, boolean wrap) {
-      super(el, wrap);
-    }
-
-    @Override
-    protected int drawUnselectedText(Graphics g, int x, int y,
-                                     int p0, int p1) throws BadLocationException {
-      final Element root = getElement();
-      return drawColoredText(g, x, y, this, getDocument(), p0, p1, root.getElement(root.getElementIndex(p0)));
-    }
-  }
-
-  private class PView extends PlainView {
-    private PView(Element el) {
-      super(el);
-    }
-
-    @Override
-    protected int drawUnselectedText(Graphics g, int x, int y,
-                                     int p0, int p1) throws BadLocationException {
-      Element root = getElement();
-      return drawColoredText(g, x, y, this, getDocument(), p0, p1, root.getElement(root.getElementIndex(p0)));
-    }
-  }
-
-  /**
-   * Determines the color with which to draw a given line of text
-   * @return the Color to draw
-   */
-  protected Color getColor(Element elem) {
-    Color col = null;
-    try {
-      final String s = elem.getDocument().getText(
-        elem.getStartOffset(), elem.getEndOffset() - elem.getStartOffset()
-      ).trim();
-
-      if (s.length() > 0) {
-        switch (s.charAt(0)) {
-          case '*':
-            col = gameMsg;
-            break;
-          case '-':
-            col = systemMsg;
-            break;
-          default:
-            if (s.startsWith(formatChat(""))) { //$NON-NLS-1$
-              col = myChat;
-            }
-            else {
-              col = otherChat;
-            }
-            break;
-        }
-      }
-    }
-    catch (BadLocationException e) {
-      ErrorDialog.bug(e);
-    }
-
-    return col == null ? Color.black : col;
-  }
-
-  /**
-   * This is a {@link Command} object that, when executed, displays
-   * a text message in the Chatter's text area     */
-  public static class DisplayText extends Command {
-    private String msg;
-    private Chatter c;
-
-    public DisplayText(Chatter c, String s) {
-      this.c = c;
-      msg = s;
-      if (msg.startsWith("<>")) {
-        msg = "<(" + Chatter.getAnonymousUserName() + ")>" + s.substring(2);
-      }
-      else {
-        msg = s;
-      }
-    }
-
-    @Override
+		if (c instanceof DisplayText) {
+			return "CHAT" + ((DisplayText) c).msg; //$NON-NLS-1$
+		} else if (c instanceof VASSAL.build.module.Chatter.DisplayText) {
+			return "CHAT" + ((VASSAL.build.module.Chatter.DisplayText) c).getMessage(); //$NON-NLS-1$	
+		} else {
+			return null;
+		}
+	}
+
+	/**
+	 * Displays the message, Also logs and sends to the server a {@link Command}
+	 * that displays this message
+	 */
+	public void send(String msg) {
```

Comment:
No change in this method except indentation, probably better to revert it to the original so it won't show up in the diff.

Or change the condition in the next line to the slightly better readable `if (!StringUtils.isEmpty(msg))`.

Reply:
fixed